docs/tutorial/values.md

```diff
@@ -0,0 +1,215 @@
+# Arr.ai values
+
+Arr.ai has three kinds of values: numbers, tuples and sets.
+
+In the following sections, enter the text appearing after each `@>` and confirm
+that the output matches what you expect.
+
+## Numbers
+
+```text
+@> 42
+42
+@> +1.23E4
+12300
+@> +1.23E40
+1.23E40
+@> -10
+-10
+```
+
+**Explore:** What do the following output?
+
+```text
+@> 1.23E-4
+@> 1.23E-5
+@> --10
+@> 1/0
+@> 0/0
+```
+
+## Tuples
+
+Tuples associate names with values. Each name/value pair is called an attribute.
+
+```text
+@> ()
+()
+@> (x: 1, y: 2)
+(x: 1, y: 2)
+@> (y: 2, x: 1)
+(x: 1, y: 2)
+@> (start: (x: 1, y: 2), end:(x: -5, y: 3))
+(end: (x: -5, y: 3), start: (x: 1, y: 2))
+```
+
+Note that tuple names are unordered. That is, `(x: 1, y: 2)` and `(y: 2, x: 1)`
+represent the same tuple. However, for convenience and consistency, arr.ai will
+usually print tuples with names ordered lexicograpically.
+
+Attribute names can be pretty much anything. Names that don't fit the standard
+identifier rules can be written using string syntax:
+
+```text
+@> ('-1-': 23, '-2-': 34)
+('-1-': 23, '-2-': 34)
+@> ('\007': 100)
+('\a': 100)
+@> ('🐭': "mouse", '🐴': "horse", '🐱': "cat")
+('🐭': 'mouse', '🐱': 'cat', '🐴': 'horse')
+```
+
+**Explore:** What do the following output?
+
+```text
+@> ((a: 1, b: 2))
+@> ((()))
+```
```

Comment:
This was interesting. My guess is that the outer parentheses are parsed as grouping operations? Might want to add explanations on `Explore` sections, maybe as the last section.

Reply:
I want there to be some element of exploration without spoon-feeding. One thought I had was to link to reference materials for explanations if people want to know more.